### PR TITLE
feat(ST-129): add update classroom api endpoint

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added Update classroom DTO
- Added PATCH `classrooms/:classroomId` api route
- Added service method
- Added repository method

## Associated Ticket Link(s)

- https://trello.com/c/mFZ7B3aL

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/classrooms/classrooms.dto.ts
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.dto.ts:1]: NIT - classrooms.dto.ts => classrooms.dtos.ts

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

File: src/classrooms/classrooms.service.ts [hunk 1]
```diff
@@ -114,4 +114,19 @@ export class ClassroomsService {
 
     return students;
   }
+
+  async updateClassroom(
+    userId: number,
+    classroomId: number,
+    updateClassroomDto: UpdateClassroomDto,
+  ) {
+    const user = await this.usersRepository.findOneOrFail({ id: userId });
```
Comment (ragib-hossain-sazim) [inline src/classrooms/classrooms.service.ts:123]: [Question] if we are already sending `classroomId` from client, then why do we need to find `user` and use `userId` to find a classroom?

Comment (faysal-alam-sazim) [reply src/classrooms/classrooms.service.ts:123]: I've added this as verification so that one teacher can't update other teacher's classroom.

Comment (ishmam-iqbal-sazim) [reply src/classrooms/classrooms.service.ts:123]: This sounds like a good use case to implement a classroom guard

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending classroom guard for update classroom endpoint.

**Pull request merged**